proto/cosmos/authz/v1beta1/authz.proto

```diff
@@ -7,21 +7,20 @@ import "gogoproto/gogo.proto";
 import "google/protobuf/any.proto";
 
 option go_package = "github.com/cosmos/cosmos-sdk/x/authz/types";
+option (gogoproto.goproto_getters_all) = false;
 
 // GenericAuthorization gives the grantee unrestricted permissions to execute
 // the provided method on behalf of the granter's account.
 message GenericAuthorization {
   option (cosmos_proto.implements_interface) = "Authorization";
 
-  // method name to grant unrestricted permissions to execute
-  // Note: MethodName() is already a method on `GenericAuthorization` type,
-  // we need some custom naming here so using `MessageName`
-  string method_name = 1 [(gogoproto.customname) = "MessageName"];
+  // Msg, identified by it's type URL, to grant unrestricted permissions to execute
+  string msg = 1;
```

Comment:
```suggestion
  string msg_type_url = 1;
```

Reply:
Or... we could alternatively consider this

```suggestion
  google.protobuf.Any msg = 1;
```

Reply:
The first suggestion won't work - see https://github.com/cosmos/cosmos-sdk/pull/9042#discussion_r619134467

Reply:
If we use `google.protobuf.Any msg = 1;`, what should its value hold? asaict, only the type_url is needed, so in order to store minimum stuff (and also not to confuse users), I prefer to keep string.

Reply:
Let's just use `string msg_type_url`

Reply:
I'm using `msg` because it allows to use `MsgTypeURL` as method name in the `Authorization` interface (which, imho is more in Go style and more "straight"). Amaury [suggested](https://github.com/cosmos/cosmos-sdk/pull/9042/files#r622043767) to use `GetMsgTypeURL`  as an alternative. If you have a preference then let me know.